src/apps/weblib/ts-api/data.ts

```diff
@@ -84,8 +84,16 @@ export class Data {
 
 		const seriesType = series.type ? series.type : this._detectType(values)
 
-		if (seriesType === 'dimension') {
-			this._addDimension(series.name, values)
+		if ('categories' in series) {
```

Comment:
I would put this condition to detectType as well, because now it's return value can be wrong. (will return measure for indexed dimensions)

Reply:
it is unnecessary to call detectType if there are categories. I would prefer to place the detect in else statement.